### PR TITLE
chore(deps): security updates

### PR DESCRIPTION
**Pull request by hpidcock**
Adds govulncheck linting to help with identifying vulnerable imported modules that are used.

Updates dependencies to fix: 
- [GO-2024-2958](https://pkg.go.dev/vuln/GO-2024-2958)
- [GO-2024-2918](https://pkg.go.dev/vuln/GO-2024-2918)
- [GO-2024-2611](https://pkg.go.dev/vuln/GO-2024-2611)

## QA steps

Build. Bootstrap. Deploy.

## Documentation changes

N/A

## Links

https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,8 @@ module github.com/juju/juju
 go 1.21
 
 require (
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.0
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.4.0
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.12.0
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.6.0
```
Comment (wallyworld) [inline go.mod:7]: Did we check bootstrap, deploy on Azure

**Review by wallyworld — APPROVED** [review]
We'll need to carefully check ci for azure breakage, or better, test before landing.

File: go.mod (hunk 1 above)
Comment (hpidcock) [reply go.mod:7]: about to do that

**Pull request merged**